### PR TITLE
2024 doc updates

### PR DESCRIPTION
**Pull request by mw-hrastega**
- Documented MATLAB build options
- Documented access to MATLAB build result
- Added notes on MATLAB Current folder
- Made enhancements

### CODE REVIEW

File: CONFIGDOC.md [hunk 1]
```diff
@@ -46,31 +46,35 @@ You can use the [`matlabroot`](https://www.mathworks.com/help/matlab/ref/matlabr
...
  
-You can customize the **Run MATLAB Tests** build step in the step configuration interface. For example, you can add source folders to the MATLAB search path, control which tests to run, and generate various test and coverage artifacts. If you do not select any of the existing options, all the tests in your project run, and any test failure causes the build to fail.
+You can customize the **Run MATLAB Tests** build step in the step configuration interface. For example, you can add source folders to the MATLAB search path, control which tests to run, and generate test and coverage artifacts. If you do not select any of the existing options, all the tests in your project run, and any test failure causes the step to fail.
```
Comment (mw-hrastega) [inline CONFIGDOC.md:77]: @sameagen-MW: Please carefully review the paragraph on line 99 w.r.t. the -sd option.

File: CONFIGDOC.md [hunk 2]
```diff
@@ -46,31 +46,35 @@ You can use the [`matlabroot`](https://www.mathworks.com/help/matlab/ref/matlabr
...
 
-MATLAB exits with exit code 0 if the specified tasks run without error. Otherwise, MATLAB terminates with a nonzero exit code, which causes the Jenkins build to fail.
+MATLAB exits with exit code 0 if the specified tasks run without error. Otherwise, MATLAB terminates with a nonzero exit code, which causes the step to fail.
 
-When you use this step, a file named `buildfile.m` must be in the root of your repository. For more information about the build tool, see [Create and Run Tasks Using Build Tool](https://www.mathworks.com/help/matlab/matlab_prog/create-and-run-tasks-using-build-tool.html).
+You can access the result of running a MATLAB build interactively in your Jenkins Interface. After your build runs, the Jenkins build summary page displays the number of tasks that ran, failed, and were skipped. If your build ran successfully, you can click the **MATLAB Build Result** link on the page to access the table of task results. This table provides information about each task that was part of the MATLAB build. Clicking a task name in the table gives you access to relevant build log information in **Console Output**.
```
Comment (sameagen-MW) [inline CONFIGDOC.md:72]: I don't think the "I" in "interface" needs to be capitalized here? Or is the "Jenkins Interface" a formal thing in the Jenkins doc?

Comment (sameagen-MW) [reply CONFIGDOC.md:72]: Also, I think we may need some words around "Console Output." Maybe "... information in the **Console Output** tab"

I don't think this affects understanding though, so no big deal if not.

File: CONFIGDOC.md (hunk 1 above)
Comment (sameagen-MW) [reply CONFIGDOC.md:77]: I think that paragraph still works, and I prefer it to the alternative of specifically calling out the behavior with `-sd`. If we get any xBug reports then we should revisit the issue and consider adding another sentence to clarify but for now I have no flags.

File: CONFIGDOC.md [hunk 3]
```diff
@@ -233,9 +239,9 @@ Use the `runMATLABBuild` step in your pipeline to run a build using the [MATLAB
...
+`buildOptions`           | <p>(Optional) MATLAB build options, specified as a list of options separated by spaces. The plugin supports the same [options](https://www.mathworks.com/help/matlab/ref/buildtool.html#mw_50c0f35e-93df-4579-963d-f59f2fba1dba) that you can pass to the `buildtool` command when running a MATLAB build.<p/><p>**Example:** `buildOptions: -continueOnFailure`<br/>**Example:** `buildOptions: -continueOnFailure -skip test`</p>
+`startupOptions`         | <p>(Optional) MATLAB startup options, specified as a list of options separated by spaces. For more information about startup options, see [Commonly Used Startup Options](https://www.mathworks.com/help/matlab/matlab_env/commonly-used-startup-options.html).</p><p>Using this input to specify the `-batch` or `-r` option is not supported.</p><p>**Example:** `startupOptions: '-nojvm'`<br/>**Example:** `startupOptions: '-nojvm -logfile "output.log"'`</p>
```
Comment (sameagen-MW) [inline CONFIGDOC.md:244]: Same comment about quotes not being necessary around `output.log`

File: CONFIGDOC.md [hunk 4]
```diff
@@ -350,8 +355,8 @@ Use the `runMATLABCommand` step in your pipeline to run MATLAB scripts, function
...
```
Comment (sameagen-MW) [inline CONFIGDOC.md:359]: Ditto with `output.log`

Comment (mw-hrastega) [reply CONFIGDOC.md:359]: Done.

File: CONFIGDOC.md (hunk 3 above)
Comment (mw-hrastega) [reply CONFIGDOC.md:244]: Done.

File: CONFIGDOC.md (hunk 2 above)
Comment (mw-hrastega) [reply CONFIGDOC.md]: Thanks for spotting the casing issue. Fixed it.

Regarding the **Console Output** part, I will consult the editor at the editorial review phase. **Console Output** appears as a tab/option name as well as a page title.

**Review by tsharmaMW — APPROVED** [review]

File: CONFIGDOC.md [hunk 5]
```diff
@@ -46,31 +46,35 @@ You can use the [`matlabroot`](https://www.mathworks.com/help/matlab/ref/matlabr
...
 
-MATLAB exits with exit code 0 if the specified tasks run without error. Otherwise, MATLAB terminates with a nonzero exit code, which causes the Jenkins build to fail.
+MATLAB exits with exit code 0 if the specified tasks run without error. Otherwise, MATLAB terminates with a nonzero exit code, which causes the step to fail.
 
-When you use this step, a file named `buildfile.m` must be in the root of your repository. For more information about the build tool, see [Create and Run Tasks Using Build Tool](https://www.mathworks.com/help/matlab/matlab_prog/create-and-run-tasks-using-build-tool.html).
+You can access the result of running a MATLAB build interactively in your . After your build runs, the Jenkins build summary page displays the number of tasks that ran, failed, and were skipped. If your build ran successfully, you can click the **MATLAB Build Result** link on the page to access the table of task results. This table provides information about each task that was part of the MATLAB build. Clicking a task name in the table gives you access to relevant build log information in **Console Output**.
```
Comment (nbhoski) [inline CONFIGDOC.md:72]: Is it correct ? "interactively in your." "After your build runs" may be we may not need in your

Comment (nbhoski) [inline CONFIGDOC.md:72]: Also Do we need screen shot to show how it might look ?

**Review by nbhoski — CHANGES_REQUESTED** [review]

File: CONFIGDOC.md (hunk 5 above)
Comment (mw-hrastega) [reply CONFIGDOC.md:72]: I don't think so. The feature is self-explanatory. Besides, this plugin configuration page contains only config-level screenshots. Result-related screenshots could lead to confusion.

Comment (mw-hrastega) [reply CONFIGDOC.md:72]: Oh, good catch. Thanks! I should have mistakenly removed "Jenkins interface." Fixed it.

File: CONFIGDOC.md [hunk 6]
```diff
@@ -233,9 +239,9 @@ Use the `runMATLABBuild` step in your pipeline to run a build using the [MATLAB
...
-
+`tasks`                   | <p>(Optional) Tasks to run, specified as a list of task names separated by spaces. If you specify the step without this input (for example, `runMATLABBuild()`),  the plugin runs the default tasks in `buildfile.m` as well as all the tasks on which they depend.</p><p>MATLAB exits with exit code 0 if the tasks run without error. Otherwise, MATLAB terminates with a nonzero exit code, which causes the step to fail.</p><p>**Example:** `tasks: 'test'`<br/>**Example:** `tasks: 'compile test'`</p>
+`buildOptions`           | <p>(Optional) MATLAB build options, specified as a list of options separated by spaces. The plugin supports the same [options](https://www.mathworks.com/help/matlab/ref/buildtool.html#mw_50c0f35e-93df-4579-963d-f59f2fba1dba) that you can pass to the `buildtool` command when running a MATLAB build.<p/><p>**Example:** `buildOptions: -continueOnFailure`<br/>**Example:** `buildOptions: -continueOnFailure -skip test`</p>
```
Comment (mcafaro) [inline CONFIGDOC.md:243]: Seems like these examples should be wrapped in quotes to be consistent with tasks and startup options.

```suggestion
`buildOptions`           | <p>(Optional) MATLAB build options, specified as a list of options separated by spaces. The plugin supports the same [options](https://www.mathworks.com/help/matlab/ref/buildtool.html#mw_50c0f35e-93df-4579-963d-f59f2fba1dba) that you can pass to the `buildtool` command when running a MATLAB build.<p/><p>**Example:** `buildOptions: '-continueOnFailure'`<br/>**Example:** `buildOptions: '-continueOnFailure -skip test'`</p>

```

**Review by mcafaro — APPROVED** [review]

File: CONFIGDOC.md (hunk 6 above)
Comment (mw-hrastega) [reply CONFIGDOC.md]: Great comment. Thanks!

**Pull request merged**